src/objective/regression_obj.cu

```diff
@@ -107,13 +109,11 @@ class RegLossObj : public ObjFunction {
             }
             out_gpair_ptr[idx] = GradientPair(Loss::FirstOrderGradient(p, label) * w,
                                               Loss::SecondOrderGradient(p, label) * w);
-            if (out_gpair_ptr[idx].GetHess() < 0.0f) {
```

Comment:
would it be better to add this check in next PRs as we can avoid extra gradient/hessian vector reading [here](https://github.com/dmlc/xgboost/blob/master/src/tree/updater_quantile_hist.cc#L458) ?

Reply:
@ShvetsKS I think all the regression objectives defined by template have this check in their respective hessian function.  Could you please verify?

Reply:
Specifically, they have something similar to `fmaxf(result, eps)` and eps is a positive value.

Reply:
seems all `SecondOrderGradient`'s are bigger than zero: https://github.com/dmlc/xgboost/blob/4224c08cacceba3f83f90e387c07aa6205a83bfa/src/objective/regression_loss.h
as all results returned with std::max(x, eps), where eps > 0;
and therefore check [here](https://github.com/dmlc/xgboost/blob/master/src/tree/updater_quantile_hist.cc#L483) is useless. So I'll just delete it in `updater_quantile_hist.cc` in next PR.

Thanks for providing the cleanup! I'll try to be more accurate.

Reply:
Also, we should probably add a check to custom objectives provided by users before making that change.